src/App.jsx

```diff
@@ -14,45 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const ALPHABETICAL_SORT = 'alphabetical';
+const LENGTH_SORT = 'length';
+
+function getSortedList(goods, { sortOrder, isReversed }) {
+  const goodsForSort = [...goods];
+
+  if (sortOrder === LENGTH_SORT) {
+    goodsForSort.sort((a, b) => a.length - b.length);
+  }
+
+  if (sortOrder === ALPHABETICAL_SORT) {
+    goodsForSort.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (isReversed) {
+    goodsForSort.reverse();
+  }
+
+  return goodsForSort;
+}
+
+export const App = () => {
+  const [sortOrder, setSortOrder] = useState('');
+  const [isReversed, setIsReversed] = useState('');
+  const sortGoodsList = getSortedList(
+    goodsFromServer, { sortOrder, isReversed },
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const switchRevers = () => setIsReversed(prevValue => !prevValue);
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortOrder('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            {
+              'is-light': sortOrder !== ALPHABETICAL_SORT,
+            },
+          )}
+          onClick={() => (
+            setSortOrder(ALPHABETICAL_SORT)
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            {
+              'is-light': sortOrder !== LENGTH_SORT,
+            },
+          )}
+          onClick={() => (
+            setSortOrder(LENGTH_SORT)
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            {
+              'is-light': !isReversed,
+            },
+          )}
```

Comment:
```suggestion
          className={cn('button is-warning', {
              'is-light': !isReversed,
            })}
```

Reply:
Do it in all cases

Reply:
@SerhiiKirik the option you suggested is less supported, so it shouldn't be changed